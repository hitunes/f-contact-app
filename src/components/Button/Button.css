.button__wrapper {
  color: rgba(0, 0, 0, 0.87);
  background-color: #ffffff;
  -webkit-transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
  transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  -webkit-box-shadow: rgba(0, 0, 0, 0.12) 0px 1px 6px, rgba(0, 0, 0, 0.12) 0px 1px 4px;
          box-shadow: rgba(0, 0, 0, 0.12) 0px 1px 6px, rgba(0, 0, 0, 0.12) 0px 1px 4px;
  border-radius: 2px;
}

.button__button {
  border: 10px;
  cursor: pointer;
  text-decoration: none;
  height: 36px;
  line-height: 36px;
  width: 100%;
  border-radius: 2px;
  -webkit-transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
  transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
  background-color: #5700A0;
  text-align: center;
}

.button__title {
  height: 36px;
  border-radius: 2px;
  -webkit-transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
  transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
  top: 0px;
}

.button__title div {
  opacity: 1;
  font-size: 14px;
  letter-spacing: 0px;
  font-weight: 500;
  margin: 0px;
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  padding-left: 16px;
  padding-right: 16px;
  color: #ffffff;
}
/*# sourceMappingURL=Button.css.map */