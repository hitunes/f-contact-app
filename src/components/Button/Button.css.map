{
    "version": 3,
    "mappings": "AACK,AAAD,gBAAU,CAAC;EACP,KAAK,EAAE,mBAAmB;EAC1B,gBAAgB,EAAE,OAAO;EACzB,UAAU,EAAE,GAAG,CAAC,KAAK,CAAC,8BAA8B,CAAC,GAAG;EACxD,UAAU,EAAE,UAAU;EACtB,UAAU,EAAE,mBAAmB,CAAC,GAAG,CAAC,GAAG,CAAC,GAAG,EAAE,mBAAmB,CAAC,GAAG,CAAC,GAAG,CAAC,GAAG;EAC5E,aAAa,EAAE,GAAG;CACrB;;AACA,AAAD,eAAS,CAAC;EACN,MAAM,EAAE,IAAI;EACZ,MAAM,EAAE,OAAO;EACf,eAAe,EAAE,IAAI;EACrB,MAAM,EAAE,IAAI;EACZ,WAAW,EAAE,IAAI;EACjB,KAAK,EAAE,IAAI;EACX,aAAa,EAAE,GAAG;EAClB,UAAU,EAAE,GAAG,CAAC,KAAK,CAAC,8BAA8B,CAAC,GAAG;EACxD,gBAAgB,EAAE,OAAO;EACzB,UAAU,EAAE,MAAM;CACrB;;AACA,AAAD,cAAQ,CAAC;EACL,MAAM,EAAE,IAAI;EACZ,aAAa,EAAE,GAAG;EAClB,UAAU,EAAE,GAAG,CAAC,KAAK,CAAC,8BAA8B,CAAC,GAAG;EACxD,GAAG,EAAE,GAAG;CAYX;;AAhBA,AAKG,cALI,CAKJ,GAAG,CAAC;EACA,OAAO,EAAE,CAAC;EACV,SAAS,EAAE,IAAI;EACf,cAAc,EAAE,GAAG;EACnB,WAAW,EAAE,GAAG;EAChB,MAAM,EAAE,GAAG;EACX,WAAW,EAAE,IAAI;EACjB,YAAY,EAAE,IAAI;EAClB,aAAa,EAAE,IAAI;EACnB,KAAK,EAAE,OAAO;CACjB",
    "sources": [
        "Button.scss"
    ],
    "names": [],
    "file": "Button.css"
}