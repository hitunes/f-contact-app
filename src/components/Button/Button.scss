.button {
    &__wrapper {
        color: rgba(0, 0, 0, 0.87);
        background-color: #ffffff;
        transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
        box-sizing: border-box;
        box-shadow: rgba(0, 0, 0, 0.12) 0px 1px 6px, rgba(0, 0, 0, 0.12) 0px 1px 4px;
        border-radius: 2px;
    }
    &__button {
        border: 10px;
        cursor: pointer;
        text-decoration: none;
        height: 36px;
        line-height: 36px;
        width: 100%;
        border-radius: 2px;
        transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
        background-color: #5700A0;
        text-align: center;
    }
    &__title {
        height: 36px;
        border-radius: 2px;
        transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
        top: 0px;
        div {
            opacity: 1;
            font-size: 14px;
            letter-spacing: 0px;
            font-weight: 500;
            margin: 0px;
            user-select: none;
            padding-left: 16px;
            padding-right: 16px;
            color: #ffffff;
        }
    }
}