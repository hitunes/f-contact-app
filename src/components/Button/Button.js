import "./Button.css";
import React from "react";

export const Button = ({ children, onClick, ...props }) => {
  return (
    <div className="button__wrapper">
      <button className="button__button">
        <div onClick={onClick} className="button__title">
          <div>{children}</div>
        </div>
      </button>
    </div>
  );
};
