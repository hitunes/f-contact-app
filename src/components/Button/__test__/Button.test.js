import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Button } from "../Button";
configure({ adapter: new Adapter() });

describe("Button", () => {
  describe("Button", () => {
    it("should be defined", () => {
      expect(Button).toBeDefined();
    });
    it("should render correctly", () => {
      const tree = shallow(<Button name="button test" />);
      expect(tree).toMatchSnapshot();
    });
  });
});
