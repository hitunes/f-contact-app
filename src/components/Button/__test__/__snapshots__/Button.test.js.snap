// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    name="button test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="button__button"
      >
        <div
          className="button__title"
        >
          <div />
        </div>
      </button>,
      "className": "button__wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="button__title"
        >
          <div />
        </div>,
        "className": "button__button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div />,
          "className": "button__title",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="button__button"
        >
          <div
            className="button__title"
          >
            <div />
          </div>
        </button>,
        "className": "button__wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="button__title"
          >
            <div />
          </div>,
          "className": "button__button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div />,
            "className": "button__title",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Button renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="button__button"
      >
        <div
          className="button__title"
        >
          <div />
        </div>
      </button>,
      "className": "button__wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="button__title"
        >
          <div />
        </div>,
        "className": "button__button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div />,
          "className": "button__title",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="button__button"
        >
          <div
            className="button__title"
          >
            <div />
          </div>
        </button>,
        "className": "button__wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="button__title"
          >
            <div />
          </div>,
          "className": "button__button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div />,
            "className": "button__title",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Button should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    name="button test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="button__button"
      >
        <div
          className="button__title"
        >
          <div />
        </div>
      </button>,
      "className": "button__wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="button__title"
        >
          <div />
        </div>,
        "className": "button__button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div />,
          "className": "button__title",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="button__button"
        >
          <div
            className="button__title"
          >
            <div />
          </div>
        </button>,
        "className": "button__wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="button__title"
          >
            <div />
          </div>,
          "className": "button__button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div />,
            "className": "button__title",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
